timesketch/lib/llms/features/interface.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2025 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Interface for LLM features."""
+
+from typing import Any, Optional
+from abc import ABC, abstractmethod
+from timesketch.models.sketch import Sketch
+
+
+class LLMFeatureInterface(ABC):
+    """Interface for LLM features."""
+
+    NAME: str = "llm_feature_interface"  # Must be overridden in subclasses
+    RESPONSE_SCHEMA: Optional[dict[str, Any]] = None
+
+    @abstractmethod
+    def generate_prompt(self, sketch: Sketch, **kwargs: Any) -> str:
+        """Generates a prompt for the LLM.
+
+        Args:
+            sketch_id: The ID of the sketch.
+            kwargs: Feature-specific keyword arguments for prompt generation.
+
+        Returns:
+            The generated prompt string.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def process_response(self, llm_response: str, **kwargs: Any) -> dict[str, Any]:
```

Comment:
It's not clear to me what processing means in this context. What is expected from the processing? Perhaps update the doc string with some use cases and/or examples.

Reply:
Done, good point